core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java

```diff
@@ -402,18 +408,25 @@ public T retry() {
          * a special case that looks for Code.NONODE in the KeeperException, then non-existence can
          * not be cached.
          */
+
         cacheWriteLock.lock();
         try {
+
           final ZooKeeper zooKeeper = getZooKeeper();
-          Stat stat = zooKeeper.exists(zPath, watcher);
+
           byte[] data = null;
+
+          boolean watched = isWatched(zooKeeper, zPath);
+
+          Stat stat = zooKeeper.exists(zPath, watched ? watcher : null);
+
           if (stat == null) {
             if (log.isTraceEnabled()) {
               log.trace("zookeeper did not contain {}", zPath);
             }
           } else {
             try {
-              data = zooKeeper.getData(zPath, watcher, stat);
+              data = zooKeeper.getData(zPath, watched ? watcher : null, stat);
```

Comment:
So in this case if we are not watched (meaning the property is either not in the list of does not exist), then we will not watch it at all?

Reply:
The default and expected value of the boolean 'watched' is true.  That is what we used to do for every node.